### PR TITLE
[ROCM] mark unit tests as working, skip failing unit test

### PR DESCRIPTION
**Pull request by iotamudelta**
* enabled fp16 tests for test_torch

* enable fp16 tests for test_nn

* enabled multilabelmargin loss for fp16

* removed skip for test_pdist_empty_col

* Enable test_nn tests that pass with compiler fixes etc.

* Enable test_legacy_nn tests that pass with compiler fixes etc.

@ezyang @bddppq

### CODE REVIEW

File: test/test_cuda.py [hunk 1]
```diff
@@ -1136,6 +1136,7 @@ def test_scatter_cpu_sizes(self):
     def test_scatter_gpu(self):
         self._test_scatter(torch.randn(4, 4).cuda(), dim=0)
 
+    @skipIfRocm
```
Comment (bddppq) [inline test/test_cuda.py]: Does this test before this PR?

**Comment by bddppq** [comment]
Triggered tests https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger/354/

@iotamudelta Before we re-enable running tests for PR, when you submit PR that affects rocm tests, please trigger the CI job here https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger/build?delay=0sec with `GIT_COMMIT` filled in with `origin/pr/XXX/head` where XXX is the PR number (in this case it's `12313`)

**Comment by bddppq** [comment]
[lint check](https://travis-ci.org/pytorch/pytorch/jobs/436950632) failed 
```
./test/test_nn.py:6965:9: E123 closing bracket does not match indentation of opening bracket's line
```

File: test/test_cuda.py (hunk 1 above)
Comment (iotamudelta) [reply test/test_cuda.py]: No, this came in broken with a recent upstream merge and is skipped here to fix CI tests.

Comment (bddppq) [reply test/test_cuda.py]: But we don't see it failing in current pytorch master

Comment (bddppq) [reply test/test_cuda.py]: By upstream you mean pytorch master or your fork?

Comment (iotamudelta) [reply test/test_cuda.py]: upstream == pytorch master

it certainly failed for us on CI, maybe it's just flaky. We have a few that behave like that and in the interest of stability I disable them.

Comment (bddppq) [reply test/test_cuda.py]: AFAICT there is no flakiness has been observed in the pass couple days in the upstream CI. Are you sure it's not some changes done in the fork that cause the failure?

**Review by bddppq — APPROVED** [review]

**Pull request closed without merging**